src/__tests__/host-component-names.test.tsx

```diff
@@ -11,10 +16,41 @@ const mockGetQueriesForElements = jest.spyOn(
 ) as jest.Mock;
 
 describe('getHostComponentNames', () => {
+  test('returns host component names from internal config', () => {
+    configureInternal({
+      hostComponentNames: { text: 'banana', textInput: 'banana' },
+    });
+
+    expect(getHostComponentNames()).toEqual({
+      text: 'banana',
+      textInput: 'banana',
+    });
+  });
+
+  test('throws when names are missing in the internal config', () => {
+    expect(() => getHostComponentNames()).toThrowErrorMatchingInlineSnapshot(`
+      "Missing host component names.
+
+      There seems to be an issue with your configuration that prevents React Native Testing Library from working correctly.
+      Please check if you are using compatible versions of React Native and React Native Testing Library."
+    `);
+  });
+
+  test('does not throw when wrapped in act after render has been called', () => {
```

Comment:
maybe we could add some explanation here as to why this test is needed wdyt ? or we could also just link the issue (but will take longer to understand for futur devs)

Reply:
Added comment and reference